### PR TITLE
Speech granite batched inference

### PR DESCRIPTION
**Pull request by avihu111**
allowing batched preprocessing + inference.
- allows collating a list of audios, and keeping track of original lengths
- calculating projector length for each audio
- verifying the number of audios matches the number of audio symbol
- bugfix in line 94 (original) - which added the sample within the while loop

### CODE REVIEW

**Comment by alex-jw-brooks** [comment]
Looks great, thanks @avihu111!

File: src/transformers/models/granite_speech/processing_granite_speech.py [hunk 1]
```diff
@@ -105,9 +111,17 @@ def _get_validated_text(self, text: Union[str, list]) -> List[str]:
...
-        if audios is not None and not isinstance(audios, torch.Tensor):
-            raise TypeError("Invalid audios provided! Audio should be a torch tensor.")
-        return audios
+        if isinstance(audios, torch.Tensor):
+            lengths = [audios.shape[-1]] * audios.shape[0]
+            return audios, lengths
+        elif isinstance(audios, list) and isinstance(audios[0], torch.Tensor):
+            lengths = [audio.shape[-1] for audio in audios]
+            padding = [max(lengths) - length for length in lengths]
+            padded = [torch.nn.functional.pad(audio, (0, pad)) for audio, pad in zip(audios, padding)]
```
Comment (alex-jw-brooks) [inline src/transformers/models/granite_speech/processing_granite_speech.py:120]: Do we need to create a feature attention mask to for the encoder to ignore pads?
